### PR TITLE
Fade pause menu background on opening and closing

### PR DESCRIPTION
**Pull request by Kasterisk**
**Brief Description of What This PR Does**

Pause menu background now fades in and out. Cleaned up the pause menu's open and close logic to reduce confusion.

**Progress Checklist**

Note: before starting this checklist the PR should be marked as non-draft.

- [x] PR author has checked that this PR works as intended and doesn't
      break existing features:
      https://wiki.revolutionarygamesstudio.com/wiki/Testing_Checklist
      (this is important as to not waste the time of Thrive team
      members reviewing this PR)
- [ ] Initial code review passed (this and further items should not be checked by the PR author)
- [ ] Functionality is confirmed working by another person (see above checklist link)
- [ ] Final code review is passed and code conforms to the 
      [styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md).

Before merging all CI jobs should finish on this PR without errors, if
there are automatically detected style issues they should be fixed by
the PR author. Merging must follow our
[styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md#git).

### CODE REVIEW

**Comment by hhyyrylainen** [comment]
Doesn't this need changes to the early multicellular editor scene as that also contains the pause menu in it?

**Comment by Kasterisk** [comment]
Fixed two signal connections relating to menu opening and closing in there.

**Comment by Kasterisk** [comment]
No idea why the build failed, never got those particular warnings before.

**Comment by hhyyrylainen** [comment]
Probably a newer version of the checking tools is used for the first time on a file you modified, so it now finds more things to request fixes for.

**Comment by Kasterisk** [comment]
What should be done? I don't think we should manually dispose Godot objects and it's just weird that this warning only occurs in this file and not the others. I think I'll probably just suppress it.

**Comment by hhyyrylainen** [comment]
I guess for some reason the checks detect that the type is ultimately derived from IDisposable? But then again I*m not sure why it isn't triggered for other classes as we have a bunch deriving from Godot types. I also thought it might be caused by the class inheriting something where we have written a Dispose method but that doesn't seem to be the case either. So I guess ignoring the warning is the right way to go here...

File: src/general/PauseMenu.tscn [hunk 1]
```diff
@@ -8,8 +8,128 @@
...
+tracks/2/keys = {
+"times": PoolRealArray( 0 ),
+"transitions": PoolRealArray( 1 ),
+"update": 0,
+"values": [ true ]
+}
+
 [node name="PauseMenu" type="Popup"]
 pause_mode = 2
+visible = true
```
Comment (hhyyrylainen) [inline src/general/PauseMenu.tscn:132]: I think this might be a mistake...

Comment (Kasterisk) [reply src/general/PauseMenu.tscn:132]: That's weird, it shows me that the node is hidden in the Godot editor however... I might have to remove this manually.

**Review by hhyyrylainen — APPROVED** [review]
Looks fine to me. I noticed one problem but it was probably a problem also with the previous code, which is that if you press the pause button in the microbe stage and then open the pause menu, closing the pause menu then unpauses the game incorrectly.

**Comment by hhyyrylainen** [comment]
Opened the issue: https://github.com/Revolutionary-Games/Thrive/issues/3319

@Kasterisk if you could take a look at that, that would be great as it was maybe caused by this PR.

**Comment by Kasterisk** [comment]
It's indeed a regression from my previous PR, this should be the fix: #3324.

**Pull request merged**